### PR TITLE
add ability for md + text reports

### PR DESCRIPTION
**Pull request by abist**
PR fixes https://github.com/microsoft/azuredatastudio/issues/18650

### CODE REVIEW

File: src/sql/workbench/api/browser/mainThreadDataProtocol.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/api/browser/mainThreadDataProtocol.ts:528]: I'd suggest making a defined type for this so you don't have to copy it over in a bunch of places

File: src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts:192]: Using indexes here is a bit odd - why aren't you using the property names (report and format)?

File: src/sql/workbench/services/tableDesigner/browser/tableDesignerPublishDialog.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/services/tableDesigner/browser/tableDesignerPublishDialog.ts:85]: Would suggest actually checking for markdown as a value. If the format is unknown it should probably log a warning but default to plaintext.

File: extensions/mssql/src/contracts.ts
Comment (Charles-Gagnon) [inline extensions/mssql/src/contracts.ts:1111]: Is there any specific reason you want to support just the report as a single value? Seems like you might as well simplify it and have it always take an object and just have the format be an optional value.

File: src/sql/workbench/api/browser/mainThreadDataProtocol.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/api/browser/mainThreadDataProtocol.ts:528]: And then I'd also suggest making the format a const enum value instead - much easier to work with and clearly defines what the expected values are.

**Comment by abist** [comment]
@Charles-Gagnon added this type based on offline discussion with @alanrenmsft and @caohai. Currently, the report is only in plain text, but this allows for adding markdown as well (in that case the return type would be an object with the script and the format type). Alan suggested using this type.

**Comment by Charles-Gagnon** [comment]
The type itself is fine, I'm just making suggestions about how to better define and use it.

File: src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts
Comment (alanrenmsft) [reply src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts:192]: does this actually work? you have to do report.content or report.format, right?

File: extensions/mssql/src/contracts.ts
Comment (alanrenmsft) [reply extensions/mssql/src/contracts.ts:1111]: i was mainly think about making it easy if the provider only have plain text report. also if we make it a union type the current STS will work fine (not a compelling reason though). I am fine removing the string type from it and update STS first to return the new expected data type.

**Comment by abist** [comment]
@alanrenmsft @Charles-Gagnon STS PR to change type - https://github.com/microsoft/sqltoolsservice/pull/1418

File: src/sql/azdata.proposed.d.ts
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:1490]: I'd suggest just naming this `PreviewReport` - the extension api typically doesn't have the concept of requests and responses.

File: src/sql/workbench/services/tableDesigner/browser/tableDesignerPublishDialog.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/services/tableDesigner/browser/tableDesignerPublishDialog.ts:34]: VS Code actually already has these defined, I'd just use theirs https://github.com/microsoft/azuredatastudio/blob/2bc6a0cd012de69504061a876070cfa43533fedc/src/vs/base/common/mime.ts

**Comment by Charles-Gagnon** [comment]
This needs a newer STS version too right?

**Comment by abist** [comment]
@Charles-Gagnon After the changes you requested, yes.

File: src/sql/azdata.proposed.d.ts
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:1490]: I was suggesting just naming this `PreviewReport`. After all - it's an object that is just describing the shape and format of the report so that makes sense to me.

File: extensions/mssql/src/contracts.ts
Comment (Charles-Gagnon) [inline extensions/mssql/src/contracts.ts:1111]: This needs to be updated as well

File: src/sql/azdata.proposed.d.ts
Comment (abist) [reply src/sql/azdata.proposed.d.ts:1490]: Following the convention for all other request types in table designer here. E.g. - `PublishChangesResult`, `DesignerEditResult` etc

Comment (Charles-Gagnon) [reply src/sql/azdata.proposed.d.ts:1490]: Ah. Ok, well not my favorite pattern for naming but I'll leave that up to you and @alanrenmsft to decide.

**Comment by abist** [comment]
@Charles-Gagnon @alanrenmsft It would be great if you can decide on the comments at once so I don't need to commit every time. Looks like the conflicting views come in often leading to unnecessary commits. I'm gonna undo the name change for now.

**Comment by Charles-Gagnon** [comment]
Waiting on the STS version update since that's required for these changes to work.

**Comment by abist** [comment]
@Charles-Gagnon it was merged - https://github.com/microsoft/sqltoolsservice/pull/1418

File: src/sql/azdata.proposed.d.ts
Comment (alanrenmsft) [inline src/sql/azdata.proposed.d.ts:1496]: please update `script` to 'report'

**Comment by Charles-Gagnon** [comment]
@abist You still need to update the STS version in this PR to actually get the functionality here though.

**Comment by alanrenmsft** [comment]
@Charles-Gagnon I am doing the STS vbump in this PR along with other table designer related changes. I added a note for Aditya in my PR. he will wait and merge with main after my PR is merged.

**Review by alanrenmsft — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**
